intrahospital_api/deployment_check.py

```diff
@@ -0,0 +1,201 @@
+from intrahospital_api import models as imodels
+from intrahospital_api import loader
+from intrahospital_api import get_api
+from intrahospital_api import update_lab_tests
+
+
+class RollBackError(Exception):
+    pass
+
+
+def batch_load_patients(since):
+    # this is all done much better after the refactor, but
+    # the below _should_ work ok
+    api = get_api()
+    data_deltas = api.data_deltas(since)
+    loader.update_from_batch(data_deltas)
+
+
+def update_patient(patient):
+    # this is all done much better after the refacot
+    # but the below should work the same as an
+    # initial lab test load
+    api = get_api()
+    hospital_number = patient.demographics_set.first().hospital_number
+    results = api.results_for_hospital_number(hospital_number)
+    update_lab_tests.update_tests(patient, results)
+
+
+def get_patients():
+    """
+    We only look at patients on the bacteraemia list as
+    we these should have lab tests
+    """
+    tags = Tagging.objects.filter(
+        value=patient_lists.Bacteraemia.tag
+    ).filter(
+        archived=False
+    )
+    return Patient.objects.filter(
+        episode__tagging__in=tags
+    )
+
+
+def get_qs(min_dt, max_dt=None):
+    """
+    Return all upstream lab tests for the patients
+    """
+    patients = get_patients()
+    lab_tests = lmodels.LabTest.objects.filter(
+        lab_test_type__istartswith="upstream"
+    ).filter(patient__in=patients)
+
+    max_updated = timezone.make_aware(datetime.datetime.min)
+    if max_dt is None:
+        max_dt = timezone.make_aware(datetime.datetime.max)
+
+    result = {}
+
+    for lab_test in lab_tests:
+        observations = lab_test.extras["observations"]
+        for observation in observations:
+            last_updated_str = observation["last_updated"]
+            if last_updated_str:
+                last_updated = serialization.deserialize_datetime(
+                    last_updated_str
+                )
+
+                if last_updated >= min_dt and last_updated <= max_dt:
+                    result[observation["observation_number"]] = lab_test
+
+                    if max_updated < last_updated:
+                        max_updated = last_updated
+
+    return result, max_updated
+
+
+def get_key(observation_number, lab_test):
+    """
+    A key is the observation number, lab test number and patient id.
+    """
+    return (
+        observation_number, lab_test.external_identifier, lab_test.patient_id,
+    )
+
+
+def get_and_clean(some_dt, max_dt=None):
+    """
+    Return the current keys of all lab tests then delete them.
+    """
+    obs_to_lab_test, max_updated = get_qs(some_dt, max_dt)
+    values = [get_key(i, v) for i, v in obs_to_lab_test.items()]
+
+    # remove duplicate lab tests
+    lab_tests = list(set(obs_to_lab_test.values()))
+    for lab_test in lab_tests:
+        lab_test.delete()
+
+    return values, max_updated
+
+
+def remove_in_everything(some_list, list_1, list_2):
```

Comment:
Is this not just a long way to spell `set(some_list).difference(list_1, list_2)` ?
https://docs.python.org/2/library/stdtypes.html#frozenset.difference